/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package rpcenv;

message Action {
   int32 action = 1;
}

message Step {
   bytes observation = 1;
   float reward = 2;
   bool done = 3;
   int32 episode_step = 4;
   float episode_return = 5;
   int32 cut_layer  = 6;
}

service RPCActorInference {
  rpc StreamingInference(Step) returns (Action) {}
}

/** model update **/
message Pull {
   int32 actor_id = 1;
}

message Model {
   bytes parameters = 1;
}

service RPCModelUpdate {
  rpc StreamingModelUpdate(Pull) returns (Model) {}
}

/** trajectory **/
message Trajectory {
    int32 actor_id = 1;
    repeated Step datas = 2;
}

message Uploaded {
    string ack = 1;
}

service UploadTrajectory {
  rpc TrajectoryUpload(Trajectory) returns (Uploaded) {}
}
